classes/submission/maps/Schema.php

```diff
@@ -1191,4 +1221,60 @@ protected function getStageFilesBySubmissions(Enumerable $submissions, array $st
             ->filterByFileStages($stageIds)
             ->getMany();
     }
+
+    /***
+     * Get detail about submission issues. Results are keyed by submission ID.
+     */
+    protected function getIssueBySubmissions(Enumerable $submissions): Enumerable
+    {
+        $submissionIds = [];
+        $submissions->each(function (Submission $submission) use (&$submissionIds) {
+            $submissionIds[] = $submission->getId();
+        });
+
+        $publications = Repo::publication()->getCollector()->filterBySubmissionIds($submissionIds)->getMany();
+
+        $issueIds = $publications->map(fn(Publication $publication) => $publication->getData('issueId'))
+            ->unique()
+            ->toArray();
+
+        $issues = Repo::issue()->getCollector()
+            ->filterByContextIds([$this->context->getId()])
+            ->filterByIssueIds($issueIds)
+            ->getMany();
+
+        return $submissions->mapWithKeys(function ($submission) use (&$issues, $publications) {
+            $submissionIssueIds = $publications
+                ->filter(fn(Publication $publication) => $publication->getData('submissionId') === $submission->getId())
+                ->map(fn(Publication $publication) => $publication->getData('issueId'))
+                ->toArray();
+
+            return [$submission->getId() => $issues->filter(fn($issue) => in_array($issue->getId(), $submissionIssueIds))];
+        });
+    }
+
+    /***
+     * Get details about the issue a submission will be published in.
```

Comment:
Add more typing to the `@return`, see https://phpstan.org/writing-php-code/phpdoc-types#array-shapes